### PR TITLE
Add error code name to status error messages

### PR DESCRIPTION
**Pull request by murgatroid99**
We frequently get questions that reference status errors, but they almost never realize that the `code` is important or know what it means. This adds the status code number and name to the error message.

### CODE REVIEW

File: packages/grpc-native-core/src/client.js [hunk 1]
```diff
@@ -53,6 +53,21 @@ var Duplex = stream.Duplex;
...
+ * @private
+ * @param {grpc~StatusObject} status The status object
+ * @return {Error} The resulting Error
+ */
+function createStatusError(status) {
+  let statusName = _.invert(constants.status)[status.code];
+  let message = `${status.code} ${status.name}: ${status.details}`;
+  let error = new Error(message);
+  error.code = status.code;
+  error.details = status.details;
```
Comment (kjin) [inline packages/grpc-native-core/src/client.js]: Should we include `error.metadata` as well?

Comment (murgatroid99) [reply packages/grpc-native-core/src/client.js]: Yes. That is actually what I meant to write there. The details string is in the message. It probably doesn't need to be in a separate field too.

**Comment by kjin** [comment]
LGTM once tests are amended to reflect new message changes

**Comment by murgatroid99** [comment]
I was lazy and didn't run the tests, so I didn't realized that I had so many tests that depended on the specific text of the error messages. I'll have to think about how to handle that.

**Comment by murgatroid99** [comment]
My solution is to add a details string to the error objects that exactly matches the original details string, and check that in the tests.

**Pull request merged**